cassiopeia/solver/GreedySolver.py

```diff
@@ -225,9 +225,12 @@ def __add_duplicates_to_tree(
         )
 
         for i in duplicate_groups:
-            new_internal_node = (
-                max([i for i in tree.nodes if type(i) == int]) + 1
-            )
+            if len(tree.nodes) == 1:
+                new_internal_node = len(duplicate_groups[i]) + 1
+            else:
+                new_internal_node = (
+                    max([n for n in tree.nodes if type(n) == int]) + 1
+                )
```

Comment:
There is, which is that if we remove internal nodes in removing duplicates, then the length of tree.nodes could be reduced to be less than the value of an existing node. Basically this could cause a naming collision. I think this definitely is an issue with the way we're naming internal nodes and we should address this, but for now I included this hotfix

Reply:
Gotcha. Yeah, it's a bit hacky but I understand why it needs to be there.